### PR TITLE
Update to liboboe 10.6.1

### PR DESCRIPTION
**Pull request by raphael-theriault-swi**
- Updated liboboe from 10.5.2 to 10.6.1.
- Removed `mode` setting and replaced it with `metricFormat`, which will automatically be set by the agent.
- Removed legacy xtrace format support as it is no longer supported by oboe.
- Renamed `xyz-mode-1` tests to `xyz` as this is now the only supported mode, with some small output readability improvement (ie. print status if `oboeInit` fails in tests).
- Bumped package version from 12.0.0 to 13.0.0-pre.1.
- Updated dev environment Dockerfile to properly install the gh cli (previously used method does not work anymore).

### CODE REVIEW

File: test/bindings.test.js [hunk 1]
```diff
@@ -33,15 +33,15 @@ const goodOptions = {
...
+  metricFormat: 1
 }
 
 describe('bindings.oboeInit()', function () {
   // Important: when only services key is provided the host is set to production by default
   // tests here always run against production collector. no way to "override" with settings.
 
   it('should initialize oboe with only a service key', function () {
-    const result = bindings.oboeInit({ serviceKey: `${env.SW_TEST_PROD_SERVICE_KEY}`, mode: 1 })
+    const result = bindings.oboeInit({ serviceKey: `${env.SW_TEST_PROD_SERVICE_KEY}`, metricFormat: 0 })
```
Comment (cheempz) [inline test/bindings.test.js:44]: Curious if `metricsFormat` needs to be passed in, which goes against the (inherited) contradiction with test description -- from quick look at the bindings.cc code it _should_ be set to the default 0 in the call to `oboe_init_options_set_defaults`.

**Review by cheempz — DISMISSED** [review]
LGTM @raphael-theriault-swi! I think you gave things a really close read and really impressed how much you figured out in such a short time :). Left a minor curiosity comment.

File: test/bindings.test.js (hunk 1 above)
Comment (raphael-theriault-swi) [reply test/bindings.test.js:44]: Just tested and it does, updating the test to reflect that and catch it if it doesn't anymore

**Review by cheempz — APPROVED** [review]
👍 thanks for the revisit @raphael-theriault-swi

**Pull request merged**